File: InterconnectBackend/Controllers/Requests/CreateVirtualMachineRequest.cs
namespace Controllers.Requests
{
    public class CreateVirtualMachineRequest
    {
        public required string Name { get; set; }
        public required uint Memory { get; set; }
        public required uint VirtualCPUs { get; set; }
        public required int BootableDiskId { get; set; }
    }
}

File: InterconnectFrontend/src/api/requests/CreateVirtualMachineRequest.ts
import BaseRequest from "./BaseRequest.ts";
import VirtualMachineEntityType from "../../models/enums/VirtualMachineEntityType.ts";

/**
 * Request object for creating a new virtual machine.
 */
interface CreateVirtualMachineRequest extends BaseRequest {
    /** Display name for the virtual machine */
    name: string;
    /** Memory allocation in MB */
    memory: number;
    /** Number of virtual CPUs to allocate */
    virtualCpus: number;
    /** Type/role of the virtual machine */
    type: VirtualMachineEntityType;
    /** ID of the bootable disk to use */
    bootableDiskId: number;
}

export default CreateVirtualMachineRequest;
